dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5956,6 +5956,256 @@ class FunctionFormatDecimal : public IFunction
     }
 };
 
+class FunctionTiDBUnHex : public IFunction
+{
+public:
+    static constexpr auto name = "tidbUnHex";
+    FunctionTiDBUnHex() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionTiDBUnHex>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!arguments[0]->isStringOrFixedString() && !arguments[0]->isNumber())
```

Comment:
> TiDB support `int` types, do we need to remove `int` in this function?

Yes, I think we should remove `int` in this function. Because TiDB only implements `evalString` function, and tiflash will convert all types to `string` type.

Reply:
> TiDB support `int` types, do we need to remove `int` in this function?

Yes, we should remove `int` in this function because tiflash will convert all data types to `string` type according to [here](https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L1722)

Reply:
ok